### PR TITLE
make all js

### PR DESCRIPTION
**Pull request by hwchase17**

### CODE REVIEW

**Comment by mayooear** [comment]
What caused the failed deployment?

**Comment by nfcampos** [comment]
Great!

File: ingest.ts [hunk 1]
```diff
@@ -3,24 +3,77 @@ import { OpenAIEmbeddings } from 'langchain/embeddings';
...
-const directoryPath = 'ingested_data';
-const rawDocs = fs.readdirSync(directoryPath)
-  .filter(file => file.endsWith('.json'))
-  .map(file => {
-    const data = fs.readFileSync(`${directoryPath}/${file}`, 'utf8');
-    const { page_content: pageContent, metadata } = JSON.parse(data);
-    return new Document({ pageContent, metadata })
+async function processFile(filePath: string): Promise<Document> {
+  return await new Promise<Document>((resolve, reject) => {
+    fs.readFile(filePath, 'utf8', (err, fileContents) => {
```
Comment (nfcampos) [inline ingest.ts:12]: As a reference for next time, use `fs/promises` instead of `fs`

File: ingest.ts [hunk 2]
```diff
@@ -3,24 +3,77 @@ import { OpenAIEmbeddings } from 'langchain/embeddings';
...
+  const docs = textSplitter.splitDocuments(rawDocs);
+  console.log("Vector store created.");
+
+  console.log("Creating vector store...");
+  /* Create the vectorstore */
+  const vectorStore = await HNSWLib.fromDocuments(docs, new OpenAIEmbeddings());
+  vectorStore.save("data")
+}
+
+(async () => {
```
Comment (nfcampos) [inline ingest.ts:76]: As a reference for next time, you can also do `run().then(() => console.log("done"))`

**Pull request merged**